buildlib/azure-pipelines-perf.yml

```diff
@@ -0,0 +1,129 @@
+- name: threshold
+  value: 5
+
+resources:
+  repositories:
+  - repository: PerfX
+    type: github
+    name: Mellanox-lab/PerfX
+    endpoint: Al3xR01
+    ref: master
+
+stages:
+  - stage: Prepare
+    jobs:
+      - job: Prepare
+        pool:
+          name: MLNX
+          demands:
+          - ucx_perf
+
+        steps:
+          - checkout: self
+            clean: true
+            fetchDepth: 10
+
+          - checkout: PerfX
+            clean: true
+            fetchDepth: 1
+
+          - bash: |
+              set -xeE
+              mkdir -p $(WorkDir)
+              mv $(Build.SourcesDirectory)/* $(WorkDir)    
+            displayName: Prepare WorkDir with code
+
+
+  - stage: Performance
+    dependsOn: Prepare
+    jobs:
+      - job: Perf
+        displayName: Performance testing
+        pool:
+          name: MLNX
+          demands:
+          - ucx_perf
+
+        steps:
+          - checkout: none
+            clean: true
+
+          - bash: |
+              printf '\n%79s\n\n' | tr ' ' =
+              echo "Triggered by repo: $(Build.Repository.Name)"
+              echo "Triggering repo URL: $(Build.Repository.Uri)"
+              echo "The branch of the triggering repo the build was queued for: $(Build.SourceBranch)"
+              echo "PR's merge message: $(Build.SourceVersionMessage)"
+              
+              # Extract commit SHAs from the PR's merge message.
+              SHA_After=$(echo $(Build.SourceVersionMessage) | awk '{print $2}')
+              SHA_Before=$(echo $(Build.SourceVersionMessage) | awk '{print $4}')
+                            
+              printf '\n%79s\n\n' | tr ' ' =
+              echo "Base SHA from Git log:       $(cd $(WorkDir)/ucx && git rev-parse HEAD^)"
+              echo "Base SHA from merge message: $SHA_Before"
+
+              printf '\n%79s\n\n' | tr ' ' =
+              echo "Latest SHA from Git log:  $(cd $(WorkDir)/ucx && git rev-parse HEAD)"
+              echo "Latest SHA from Azure:    $(Build.SourceVersion)"
+              echo "Topic commit to merge:    $SHA_After"
+              echo -e "The merge commit will be used for testing. \n"
+            displayName: Debug info
+            workingDirectory: $(WorkDir)
+
+          - bash: |
+              set -eE
+              # Extract commit SHAs from the PR's merge message.
+              # SHA_After=$(Build.SourceVersion)
+              SHA_After=$(echo $(Build.SourceVersionMessage) | awk '{print $2}')
+              SHA_Before=$(echo $(Build.SourceVersionMessage) | awk '{print $4}')
+
+              python3 -m venv VENV
+              source VENV/bin/activate
+              cd $(WorkDir)/PerfX
+              pip3 install -q -r requirements.txt
+              module load /hpc/local/etc/modulefiles/hpcx-gcc
+
+              set -xeE -o pipefail
+              python3 -u perfx.py \
+                --config ucx-rdmz.yml \
+                --before "$SHA_Before" \
+                --after "$SHA_After" \
```

Comment:
Since PerfX already have before/after, do you think it make sense to introduce something like `--before master --after HEAD`? I assume it should even work our of the box.

Reply:
Discovering those flags was a lifesaver!
It's great as it is if you ask me.

Reply:
I just mean we don't need to define `SHA`s like that, we can use existing functionality already. YAML can become shorter :)

Reply:
You mean, just put master and HEAD into yaml? Can try that.
Please note, that I'm using the merge topic commit instead of HEAD. Not that it should be a difference.

Reply:
Yes, that is what I mean.

Why do you use merge topic and and just something latest stable? Aren't we supposed to measure PR effect?

Reply:
Debatable. This merge commit represents the state of the code after the two branches were merged, so it should best reflect the "After" state. In fact, I see no diff between the merge commit and the last one.

Reply:
The merge base is not always 'master'. It might be v.*.*, so better to keep it dynamic instead of hardcoding in yaml.

Reply:
OK, I see. For clarity I'd rely on some specific variables here instead of merge message. There should be something like `$targetBranch` and `$currentCommit`.

Reply:
Following an offline discussion with Yossi, agreed to rely on the merge commit, as I do now.
I found no predefined var in Azure for it, thus keeping the current implementation.